### DISCUSSION TITLE
Blogeintrag Woche 06 (Sem4) - Metriken

### DISCUSSION THREAD

**Discussion by niklaskrauth** [mausio/corgi-shop-doc#16] [Blogs]
## Metriken
Das Backend wurde mit IntelliJ und dem Plugin MetricsTree im Main-Branch, um seine Qualität und Wartbarkeit zu bewerten. Der Maintainability Index liegt bei 16,1071, was als akzeptabel eingestuft wird. Dies deutet darauf hin, dass das System eine grundlegende Wartbarkeit aufweist, aber möglicherweise Raum für Verbesserungen besteht.

Die Effektivität, gemessen mit einer Metrik von 1,2813, bewertet die Nutzung von objektorientierter Programmierung (OOP). Ein Wert in diesem Bereich zeigt an, dass die Prinzipien der OOP größtenteils umgesetzt werden, was positiv zur Struktur und Erweiterbarkeit des Codes beiträgt.

Mit einer Reusability-Metrik von 1,4052 zeigt das System ein gutes Maß an Wiederverwendbarkeit von Klassen und Funktionen. Dies ist ein Hinweis darauf, dass Komponenten modular und flexibel genug gestaltet sind, um in verschiedenen Kontexten wiederverwendet zu werden.

Allerdings gibt es bei der Verständlichkeit des Codes deutliche Defizite. Der Wert liegt bei -2,5699, was auf erhebliche Schwierigkeiten bei der Nachvollziehbarkeit des Codes hindeutet. Dies könnte durch komplexe Logik, unzureichende Kommentare oder schlecht benannte Variablen und Funktionen verursacht sein.

Zusammenfassend zeigt das Backend Akzeptanz in der Wartbarkeit und weist gute Werte in Effektivität und Wiederverwendbarkeit auf. Allerdings sollte die Verständlichkeit des Codes dringend verbessert werden, um die Gesamtqualität und langfristige Wartbarkeit des Systems zu erhöhen.

## Projektfortschritt
Dazu wurde im Backend die User-Klasse komplett implementiert mit einer Basic Auth Authentifizierung, die Cart-Klasse wurde komplett in den Main-Branch hinzugefügt und UUID's wurden für die Corgi-Entitäten und Item-Entitäten erstellt.

**Comment by ghost** [comment]
Wie habt ihr vor, die Verständlichkeit des Codes zu verändern? Seht ihr in einem der genannten Aspekte (Kommentare, Logik, Variablen) mehr Spielraum als in einem anderen? 

Es ist sehr erfreulich, dass euer Projekt Fortschritte verzeichnet sowie, dass das Backend in durch die Beurteilungen der Metriken so gut abgeschnitten hat.

**Comment by ghost** [comment]
Ist toll zu sehen, dass ihr die Metriken gefunden und bewertet habt.

An sich hört sich das gut an, ich finde nur das mir das genaue Verhältnis zwischen den Zahlen fehlt.
Gibt es da vielleicht eine Skala, oder wisst ihr wie das berechnet wird?

**Comment by lukashert** [comment]
Der Blogeintrag bietet einen klaren Überblick über die aktuellen Metriken und zeigt sowohl Stärken als auch Verbesserungspotenziale auf. Schön weiter so! 👍🙌
